mlir/lib/Dialect/Rock/Transforms/GemmToGridwise.cpp

```diff
@@ -166,17 +187,55 @@ GemmRewritePattern::matchAndRewrite(GemmOp op, GemmOpAdaptor adaptor,
   if (!gridSize)
     return op.emitOpError("grid size must be set at lowering");
 
+  auto aElementType = a.getType().cast<MemRefType>().getElementType();
+  auto bElementType = b.getType().cast<MemRefType>().getElementType();
+  auto cElementType = c.getType().cast<MemRefType>().getElementType();
+
+  auto accumulatorElementType = obtainAccumulatorElementType(
+      rw, aElementType, bElementType, cElementType);
+  auto accumulatorShape = c.getType().cast<MemRefType>().getShape();
+  auto accumulatorType =
+      MemRefType::get(accumulatorShape, accumulatorElementType);
+  auto accumulator = rw.create<memref::AllocOp>(loc, accumulatorType);
```

Comment:
Let's make this conditional on whether the accumulator type is different from the output type.

Reply:
done